### PR TITLE
CDAP-6032 Cleanup files on master when program is launched.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-6032

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java [hunk 1]
```diff
@@ -109,9 +111,16 @@
...
   @Nullable
   @Override
-  protected RuntimeInfo createRuntimeInfo(ProgramController controller, ProgramId programId) {
+  protected RuntimeInfo createRuntimeInfo(ProgramController controller, ProgramId programId,
+                                          final Runnable cleanUpTask) {
     if (controller instanceof AbstractTwillProgramController) {
       RunId twillRunId = ((AbstractTwillProgramController) controller).getTwillRunId();
+      controller.addListener(new AbstractListener() {
+        @Override
+        public void alive() {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: You might want to capture the `init` method as well as the state can already be `ALIVE`.

Comment (sagarkapare) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: Added for `init` as well.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java [hunk 2]
```diff
@@ -109,9 +111,25 @@
...
+                                          final Runnable cleanUpTask) {
     if (controller instanceof AbstractTwillProgramController) {
       RunId twillRunId = ((AbstractTwillProgramController) controller).getTwillRunId();
+      controller.addListener(new AbstractListener() {
+        @Override
+        public void init(ProgramController.State currentState, @Nullable Throwable cause) {
+          if (currentState == ProgramController.State.ALIVE) {
+            alive();
+          } else {
+            super.init(currentState, cause);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: no need to call super.

**Comment by sagarkapare** [comment]
@chtyim Addressed comments. please take a look again. thanks!

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java (hunk 2 above)
Comment (sagarkapare) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: fixed.

**Comment by sagarkapare** [comment]
Build is running here - https://builds.cask.co/browse/CDAP-RUT1433-1

**Comment by chtyim** [comment]
LGTM. Please squash before merging.

**Comment by sagarkapare** [comment]
Build is green here - https://builds.cask.co/browse/CDAP-RUT1433-1. Squashing and merging.

**Pull request merged**